### ISSUE TITLE
Attempt to implement hotwire for form validation

### ISSUE THREAD

**Issue by peetucket** [sul-dlss/happy-heron#786]
Spike to see if hotwire is a good improvement for form validations.  Try to implement on item deposit page.

https://hotwire.dev/

**Comment by mjgiarlo** [comment]
# Spike report

This PR begins by [installing the hotwire-rails gem](https://github.com/sul-dlss/happy-heron/pull/820/files#diff-d09ea66f8227784ff4393d88a19836f321c915ae10031d16c93d67e6283ab55fR59), the rationale for which is above. The hotwire-rails gem rolls up two other gems, turbo-rails and stimulus-rails. These contain the javascript that's needed on the client side to make the hotwire approach functional and they also contain handy helper methods for using in views, models, controllers, etc. 

A potential downside is that they require sprockets for the asset pipeline: H2 doesn't currently use sprockets for assets (it's pure webpacker). No problem, it's easy to enable [sprockets](https://github.com/sul-dlss/happy-heron/pull/820/files#diff-c1fd91cb1911a0512578b99f657554526f3e1421decdb9e908712beab57e10f9R16) for [JavaScript assets](https://github.com/sul-dlss/happy-heron/pull/820/files#diff-12fd523cb95d119382c3743591e40c90ae62ed20b24a7012b37aed54031009b8R3).

When sprockets is turned on for JavaScript assets, the Stimulus controllers that fire in the feature specs stop working. No big problem: we can copy/move the Stimulus controllers from `app/javascript/controllers/` to `app/assets/javascripts/controllers/`.

Apparently the current stimulus-rails autoloader doesn't know how to deal with Stimulus controllers that have dashes in their name ([pending PR 1](https://github.com/hotwired/stimulus-rails/pull/3) [pending PR 2](https://github.com/hotwired/stimulus-rails/pull/11)). At any rate, we can patch that pretty easily in https://github.com/hotwired/stimulus-rails/blob/v0.1.2/app/assets/javascripts/stimulus/loaders/autoloader.js#L10 by changing that line to:
```javascript
     let controllerFilename = `${controllerName.replaceAll('-', '_')}_controller`
```

At this point, the Stimulus controllers still aren't wired up correctly as target fields cannot be found:

```
error: unable to find target for subtypeField                                                                                                         
error: unable to find target for subtypeField                                                                                                         
error: unable to find target for abstractField                                                                                                        
error: unable to find target for contact_emailField                                                                                                   
error: unable to find target for abstractField                                                                                                        
error: unable to find target for contact_emailField                                                                                                   
error: unable to find target for published_edtfField                                                                                                  
error: unable to find target for abstractField                                                                                                        
error: unable to find target for contact_emailField                                                                                                   
error: unable to find target for published_edtfField                                                                                                  
error: unable to find target for abstractField                                                                                                        
error: unable to find target for contact_emailField         
```

Also, the file upload controller that depends on Dropzone cannot find the dependency; the Sprockets-based asset pipeline wants DZ to be located in `app/assets/javascripts/libraries/`. That is easy enough to copy or move. But, per the [stimulus-rails README](https://github.com/hotwired/stimulus-rails#usage), "[t]he libraries must be made for ESM" and it is not clear that that is the case.

Long story short: apparently, in order to start making use of hotwire in H2, first we need to totally rejigger our Stimulus work to deliver it via Sprockets vs. our currently working webpacker delivery. The question is: is it worth the time?

**Comment by mjgiarlo** [comment]
Discussed :point_up: with @sul-dlss/infrastructure-team:

> @jcoyne :speech_balloon: I read your comment. I guess like if you’re hitting this much resistance this early, perhaps it’s not the right path. I would like to look into not using the rails-hotwire gem and seeing if we could just add “turbo.” But I’m not sure if it’s worth it for this project.
> @mjgiarlo :speech_balloon: Agree. Very easy to merely upgrade to Turbo, but not sure it gets H2 very much.
> @peetucket Seems viable for future projects though

So we're closing this issue and closing #820 and declaring the spike over for sprint 0.
